numba/np/polynomial/polynomial_core.py

```diff
@@ -0,0 +1,211 @@
+from numba.extending import (models, register_model, type_callable,
+                             unbox, NativeValue, make_attribute_wrapper, box,
+                             lower_builtin)
+from numba.core import types, cgutils
+from numpy.polynomial.polynomial import Polynomial
+from contextlib import ExitStack
+import numpy as np
+from llvmlite import ir
+
+
+@register_model(types.PolynomialType)
+class PolynomialModel(models.StructModel):
+    def __init__(self, dmm, fe_type):
+        members = [
+            ('coef', fe_type.coef),
+            ('domain', fe_type.domain),
+            ('window', fe_type.window)
+            # Introduced in NumPy 1.24, maybe leave it out for now
+            # ('symbol', types.string)
+        ]
+        super(PolynomialModel, self).__init__(dmm, fe_type, members)
+
+
+@type_callable(Polynomial)
+def type_polynomial(context):
+    def typer(coef, domain=None, window=None):
```

Comment:
I think the polynomial class also need to be marked as experimental. Numba ships with a `NumbaExperimentalFeatureWarning` you can emit in this case. See:
https://github.com/numba/numba/blob/f4c4afcb180193064a0c985246c7c71d007c6b1d/numba/core/utils.py#L643-L644